### PR TITLE
Move to subdirectory format

### PR DESCRIPTION
**Pull request by jeffmendoza**
@remi @bshaffer 

What do you think about the directory name for the current example: "storage"? It is currently being used here: https://cloud.google.com/storage/docs/authentication

### CODE REVIEW

**Comment by jeffmendoza** [comment]
bump

**Comment by jeffmendoza** [comment]
Updated storage example, no lib/storage. Keeping tests in spec/ as that is where "rake spec" looks

File: .travis.yml [hunk 1]
```diff
@@ -5,12 +19,17 @@ rvm:
...
+    - vendor/bundle
 
 env:
-- PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json #Other environment variables on same line
+- TEST_DIR=storage
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+- TEST_DIR=pubsub
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
```
Comment (bshaffer) [inline .travis.yml]: This is interesting to use the build matrix for each sample's test suite. Wouldn't it be simpler to run them sequentially using one build?

Comment (jeffmendoza) [inline .travis.yml]: I saw this idea and thought it was cool =)

Comment (bshaffer) [inline .travis.yml]: While I can't say I'm a huge fan (it'll complicate the build UI by adding lots of rows), I'm not hugely against it. However, you should use the `global` keyword for env:

```yaml
env:
  global:
    - PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
    - GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
  matrix:
    - TEST_DIR=storage
    - TEST_DIR=pubsub
```

Comment (jeffmendoza) [inline .travis.yml]: Ah, nice I'll do that.

Let's try the matrix, I think I prefer it to some sort of script that loops
through the dirs.

Comment (bshaffer) [inline .travis.yml]: SGTM!

File: pubsub/spec/test_test.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "spec_helper"
+
+RSpec.describe "Pub/Sub test test" do
```
Comment (remi) [inline pubsub/spec/test_test.rb]: Can we remove the `pubsub` subdirectory from this pull request?

**Comment by bshaffer** [comment]
@jeffmendoza also it looks like `list_buckets.rb` appears twice. I am assuming the one in [`lib/storage`](https://github.com/GoogleCloudPlatform/ruby-docs-samples/tree/jlm/subdirs/lib/storage) should be removed?

**Comment by remi** [comment]
`lib/storage` will be removed after the cloud.google.com code snippets have been updated to point to the new file locations :smile:

**Comment by bshaffer** [comment]
bump :)

**Comment by jeffmendoza** [comment]
Sorry, I've been at kubecon. I'll take a look tomorrow.

File: pubsub/spec/test_test.rb (hunk 2 above)
Comment (jeffmendoza) [inline pubsub/spec/test_test.rb]: I just wanted to make sure that the travis setup actually worked with more than one directory. I'll submit the PR with the pub sub sample I have right away.

**Comment by jeffmendoza** [comment]
Updated, can we merge?

**Comment by bshaffer** [comment]
@jeffmendoza I believe @remi wanted to remove the pubsub directory since no samples actually exist in there. Other than that, LGTM

**Pull request merged**